docs/guides/javascript/reactive/index.md

```diff
@@ -0,0 +1,1382 @@
+
+Let's take a look at the component anatomy.
+
+### Extending the BaseComponent class
+
+All reactive UIs are object-oriented and all components extend the BaseComponent class. This means that the default component structure is:
+
+<details>
+  <summary>View example</summary>
+  <div>
+
+```js title="path/to/plugin/amd/src/yourcomponent.js"
+import {BaseComponent} from 'core/reactive';
+
+export default class extends BaseComponent {
+(...)
+}
+```
+
+  </div>
+</details>
+
+### Instantiate a Component
+
+Any component belongs to one DOM element and is registered in a Reactive Instance. This means that to create a new instance we need to provide at least those two elements. For example:
+
+<details>
+  <summary>View example</summary>
+  <div>
+
+```js title="path/to/plugin/amd/src/main.js"
+import {reactiveInstance} from 'YOUR_PLUGIN/reactive';
+import {YourComponent} from 'YOUR_PLUGIN/yourcomponent;
+
+(...)
+
+return new YourComponent({
+     element: document.getElementById(DOM_ELEMENT_ID),
+     reactive: reactiveInstance,
+ (... other data your component needs ...)
+});
+```
+
+  </div>
+</details>
+
+In case no Reactive Instance is passed, the BaseComponent will try to inherit the instance from a parent DOM element. This way subcomponents can omit that param.
+
+### Initialize a component from a mustache template
+
+Components are easy to embed in mustache files. To do so your class must have a static "init" method that could be called inside the {{#JS}}.
+
+<Tabs>
+  <TabItem value="component-view" label="Component class" default>
+
+```js title="path/to/plugin/amd/src/childcomponent.js"
+import {BaseComponent} from 'core/reactive';
+import {reactiveInstance} from 'YOUR_PLUGIN/reactive';
+
+export default class extends BaseComponent {
+    static init(target, selectors) {
+        return new Component({
+            element: document.getElementById(target),
+            reactive: reactiveInstance,
+            selectors,
+        });
+    }
+    // (...)
+}
+```
+
+  </TabItem>
+  <TabItem value="mustache-view" label="Mustache file">
+
+```handlebars title="path/to/plugin/templates/yourthing/childcomponent.mustache"
+<div id="{{uniqid}}-your-plugin-{{id}}"> (...) </div>
+{{#js}}
+require(['YOUR_PLUGIN/local/yourthing/childcomponent'], function(component) {
+    component.init('{{uniqid}}-your-plugin-{{id}}');
+});
+{{/js}}
+```
+
+  </TabItem>
+</Tabs>
+
+The init method has an id as a first param to find the main DOM Element, but it also has a second one called `selectors`. This second param is an optional (but recommended) param any component can use. We will see how to use the selectors later but they are CSS selectors to find inner DOM elements in the main element. Bypassing as an init-param, any mustache can customize the default CSS selectors, which means the component is more reusable.
```

Comment:
I'd also put this into an `::: info` admonition. It is relevant, but will be discussed in more detail later on. This mention is just a "Heads-up" to let you the reader know that it will be covered in more detail later so hold that quesiton til then.

Reply:
Done.